### PR TITLE
feat: Add print support for orders

### LINKED ISSUE #1016: Adding printing support for Orders
### Feature Request

**Feature Description**
There should be an option to print the orders.

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by sridharjajoo**
Fixes #1016 
Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream development branch.

Changes: 
Print adapter was added for printing orders.

Screenshots for the change:
ezgif com-resize 26

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Print QR code with it

**Comment by iamareebjamal** [comment]
Create an issue to show the attendees related to order in order details

**Comment by sridharjajoo** [comment]
@iamareebjamal I think a P.R exists for it [link](https://github.com/fossasia/open-event-orga-app/pull/1127)

**Comment by iamareebjamal** [comment]
OK, no need of QR code right now as there are multiple attendees with an order

**Comment by sridharjajoo** [comment]
@iamareebjamal Kindly review.

**Comment by iamareebjamal** [comment]
Yes

File: app/src/main/java/org/fossasia/openevent/app/core/orders/detail/OrderDetailFragment.java [hunk 1]
```diff
@@ -72,6 +76,20 @@ public void onStart() {
         orderDetailViewModel.getOrder(orderIdentifier, eventId, false).observe(this, this::showOrderDetails);
         orderDetailViewModel.getProgress().observe(this, this::showProgress);
         orderDetailViewModel.getError().observe(this, this::showError);
+        binding.printAction.setOnClickListener(view -> {
+            doPrint();
+        });
+    }
+
+    private void doPrint() {
+        PrintManager printManager;
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/orders/detail/OrderDetailFragment.java]: No need of this

File: app/src/main/java/org/fossasia/openevent/app/core/orders/detail/OrderDetailFragment.java [hunk 2]
```diff
@@ -72,6 +76,20 @@ public void onStart() {
...
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            printManager = (PrintManager) getActivity().getSystemService(Context.PRINT_SERVICE);
+            String jobName = this.getString(R.string.app_name) + " Document";
+            printManager.print(jobName, new OrderDetailsPrintAdapter(getActivity(), order), null);
+        } else {
+            ViewUtils.showSnackbar(binding.getRoot(), "No Printing Support!");
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/orders/detail/OrderDetailFragment.java]: FAB shouldn't be shown if there is no printing support

**Comment by Masquerade0097** [comment]
@sridharjajoo I don't think our server supports sending email with order receipt to any entered email.

**Comment by sridharjajoo** [comment]
@Masquerade0097 We can just send the receipt of the order via an email intent that will be handled in-app.

**Comment by Masquerade0097** [comment]
@sridharjajoo EventBrite uses `orders@eventbrite.com` to send the emails related to orders to customers. Yes, we can send email using Organizer private id but to attendees, the email won't look official since there will be no official eventyay email. IMO it would be better to implement this functionality in Server first.

**Comment by sridharjajoo** [comment]
@Masquerade0097 Ok that seems right.
@iamareebjamal Should I first create an issue in the server for implementing this feature or go forward with the previous approach?

**Comment by sridharjajoo** [comment]
@iamareebjamal I have done the required changes.

File: app/src/main/java/org/fossasia/openevent/app/core/orders/detail/OrderDetailFragment.java [hunk 3]
```diff
@@ -72,6 +76,19 @@ public void onStart() {
...
+        });
+    }
+
+    private void doPrint() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+            PrintManager printManager = (PrintManager) getActivity().getSystemService(Context.PRINT_SERVICE);
+            String jobName = this.getString(R.string.app_name) + " Document";
+            printManager.print(jobName, new OrderDetailsPrintAdapter(getActivity(), order), null);
+        } else {
+            binding.printAction.setVisibility(View.GONE);
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/core/orders/detail/OrderDetailFragment.java]: Not a good idea to hide the FAB once someone clicks on the button

**Comment by sridharjajoo** [comment]
@iamareebjamal Kindly review

**Comment by iamareebjamal** [comment]
Resolve conflicts

**Comment by sridharjajoo** [comment]
@iamareebjamal Resolved the conflicts

**Pull request merged**